plans/phase-1/final-cr-definition.md

```diff
@@ -0,0 +1,845 @@
+          podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/component: multiorch
+            topologyKey: "kubernetes.io/hostname"
+    resources:
+      requests:
+        cpu: "100m"
+        memory: "128Mi"
+      limits:
+        cpu: "200m"
+        memory: "256Mi"
+
+  # --- Template for MultiGateway ---
+  multigateway:
+    replicas: 2
+    affinity:
+      podAntiAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+        - weight: 100
+          podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/component: multigateway
+            topologyKey: "kubernetes.io/hostname"
+        - weight: 50
+          podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/component: multigateway
+            topologyKey: "topology.kubernetes.io/zone"
+    resources:
+      requests:
+        cpu: "500m"
+        memory: "512Mi"
+      limits:
+        cpu: "1"
+        memory: "1Gi"
+
+  # --- Template for MultiAdmin ---
+  multiadmin:
+    replicas: 1
+    affinity:
+      podAntiAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+        - weight: 100
+          podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/component: multiadmin
+            topologyKey: "kubernetes.io/hostname"
+    resources:
+      requests:
+        cpu: "100m"
+        memory: "128Mi"
+      limits:
+        cpu: "200m"
+        memory: "256Mi"
+  
+  # --- Template for all multigres images ---    
+  images:
+    imagePullPolicy: "IfNotPresent"
+    imagePullSecrets:
+      - name: "my-registry-secret"
+    multigateway: "multigres/multigres:latest"
+    multiorch: "multigres/multigres:latest"
+    multipooler: "multigres/multigres:latest"
+    multiadmin: "multigres/multigres:latest"
+    postgres: "postgres:15.3"
+
+  managedTopoServer:
+    image: quay.io/coreos/etcd:v3.5.17
+    replicas: 3
+    affinity:
+      podAntiAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+        - weight: 100
+          podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/component: topo-server
+            topologyKey: "kubernetes.io/hostname"
+        - weight: 50
+          podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/component: topo-server
+            topologyKey: "topology.kubernetes.io/zone"
+    dataVolumeClaimTemplate:
+      accessModes: ["ReadWriteOnce"]
+      resources:
+        requests:
+          storage: "10Gi"
+ status:
+  # This list is populated by the MultigresCluster controllers using it.
+  # It shows which clusters are actively referencing this template.
+  # This can be used by a validating webhook to block deletion
+  # of the template while it is in use.
```

Comment:
The status field is never meant to be used for controller logic, and is only intended for providing human friendly details. While this status input can be provided here, it would be for presentation usage only, and the actual validating webhook logic will need to check the deletion request and its resource ID, and list all the MultigresCluster resources and see if there is any that depend on this template.